### PR TITLE
Adds `Promise.try`

### PR DESCRIPTION
**Pull request by mhassan1**
This PR adds a polyfill for `Promise.try`.

### CODE REVIEW

**Comment by mhassan1** [comment]
@romainmenke Please take a look, if you have some time.

File: polyfills/Promise/try/polyfill.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+(function () {
+	var supportsDefiningFunctionName = (function () {
+		var fn = function () {};
+		try {
+			Object.defineProperty(fn, "name", {
+				value: "test"
+			});
+			return true;
+		} catch (ignore) {
```
Comment (romainmenke) [inline polyfills/Promise/try/polyfill.js:43]:
```suggestion
		} catch (_) {
```

**Review by romainmenke — APPROVED** [review]
Thank you. LGTM 👍

-----

One tiny nit, but feel free to merge as is :)

File: polyfills/Promise/try/polyfill.js (hunk 1 above)
Comment (mhassan1) [reply polyfills/Promise/try/polyfill.js:43]: It auto-merged 😄. It looks like we have `catch (ignore)` in a lot of places, anyway.

**Pull request merged**